adal/src/main/java/com/microsoft/aad/adal/Oauth2.java

```diff
@@ -241,6 +265,13 @@ public String buildRefreshTokenRequestMessage(String refreshToken)
                     StringExtensions.urlFormEncode(mRequest.getResource()));
         }
 
+        // sending redirect uri for the refresh token request if it's provided
+        if (!StringExtensions.isNullOrBlank(mRequest.getRedirectUri())
```

Comment:
can we do a quick test about what happens if redirecturi is garbage for the non-broker scenario?

Reply:
@piqiums Have you run this test? Please sync with @addev-ashish on the outcome